capsules/src/net/thread/tlv.rs

```diff
@@ -0,0 +1,1315 @@
+    Response = 4,
+    LinkLayerFrameCounter = 5,
+    // LinkQuality = 6,         // TLV type not used in Thread
+    // NetworkParameter = 7,    // TLV type not used in Thread
+    MleFrameCounter = 8,
+    /*
+    TODO: Not required to implement MLE for SED
+    Route64 = 9,
+    */
+    Address16 = 10,
+    LeaderData = 11,
+    NetworkData = 12,
+    TlvRequest = 13,
+    ScanMask = 14,
+    Connectivity = 15,
+    LinkMargin = 16,
+    Status = 17,
+    Version = 18,
+    /*
+    TODO: Not required to implement MLE for SED
+    AddressRegistration = 19,
+    Channel = 20,
+    PanId = 21,
+    ActiveTimestamp = 22,
+    PendingTimestamp = 23,
+    */
+    ActiveOperationalDataset = 24,
+    PendingOperationalDataset = 25,
+    /*
+    TODO: Not required to implement MLE for SED
+    ThreadDiscovery = 26,
+    */
+    NotPresent,
+}
+
+impl From<u8> for TlvType {
+    fn from(tlv_type: u8) -> Self {
+        match tlv_type {
+            0 => TlvType::SourceAddress,
+            1 => TlvType::Mode,
+            2 => TlvType::Timeout,
+            3 => TlvType::Challenge,
+            4 => TlvType::Response,
+            5 => TlvType::LinkLayerFrameCounter,
+            8 => TlvType::MleFrameCounter,
+            10 => TlvType::Address16,
+            11 => TlvType::LeaderData,
+            12 => TlvType::NetworkData,
+            13 => TlvType::TlvRequest,
+            14 => TlvType::ScanMask,
+            15 => TlvType::Connectivity,
+            16 => TlvType::LinkMargin,
+            17 => TlvType::Status,
+            18 => TlvType::Version,
+            24 => TlvType::ActiveOperationalDataset,
+            25 => TlvType::PendingOperationalDataset,
+            _ => TlvType::NotPresent,
+        }
+    }
+}
+
+impl<'a, 'b> From<&'a Tlv<'b>> for TlvType {
+    fn from(tlv: &'a Tlv<'b>) -> Self {
+        match *tlv {
+            Tlv::SourceAddress(_) => TlvType::SourceAddress,
+            Tlv::Mode(_) => TlvType::Mode,
+            Tlv::Timeout(_) => TlvType::Timeout,
+            Tlv::Challenge(_) => TlvType::Challenge,
+            Tlv::Response(_) => TlvType::Response,
+            Tlv::LinkLayerFrameCounter(_) => TlvType::LinkLayerFrameCounter,
+            Tlv::MleFrameCounter(_) => TlvType::MleFrameCounter,
+            Tlv::Address16(_) => TlvType::Address16,
+            Tlv::LeaderData { .. } => TlvType::LeaderData,
+            Tlv::NetworkData(_) => TlvType::NetworkData,
+            Tlv::TlvRequest(_) => TlvType::TlvRequest,
+            Tlv::ScanMask(_) => TlvType::ScanMask,
+            Tlv::Connectivity { .. } => TlvType::Connectivity,
+            Tlv::LinkMargin(_) => TlvType::LinkMargin,
+            Tlv::Status(_) => TlvType::Status,
+            Tlv::Version(_) => TlvType::Version,
+            Tlv::ActiveOperationalDataset(_) => TlvType::ActiveOperationalDataset,
+            Tlv::PendingOperationalDataset(_) => TlvType::PendingOperationalDataset,
+        }
+    }
+}
+
+#[repr(u8)]
+pub enum LinkMode {
```

Comment:
I think the enum and field names are pretty implicit as-is. Especially if you're reading the spec along side the code.